src/hooks/data/compounders/useCompounder.ts

```diff
@@ -0,0 +1,110 @@
+import { useContractReads } from "wagmi"
+
+import {
+  findApiCompounderVaultForAsset,
+  findApiTokenVaultForAsset,
+} from "@/lib/findApiVaultForAsset"
+import { VaultProps } from "@/lib/types"
+import { useApiCompounderVaults, useApiTokenVaults } from "@/hooks/api"
+import useRegistryContract from "@/hooks/useRegistryContract"
+import {
+  useIsCurveCompounder,
+  useIsTokenCompounder,
+} from "@/hooks/useVaultTypes"
+
+export function useCompounder({ asset, type }: VaultProps) {
+  // const chainId = useActiveChainId()
+  const isCurve = useIsCurveCompounder(type)
+  const isToken = useIsTokenCompounder(type)
+
+  // Preferred: API request
+  const apiCompounderQuery = useApiCompounderVaults({ type })
+  const apiTokenQuery = useApiTokenVaults({ type })
+
+  // Fallback: contract requests
+  const enableFallback = isToken
+    ? apiTokenQuery.isError
+    : apiCompounderQuery.isError
+  const registryContract = useRegistryContract()
+  const compounderInfo = useContractReads({
+    contracts: [
+      {
+        ...registryContract,
+        functionName: isToken
+          ? "getTokenCompounderName"
+          : isCurve
+          ? "getCurveCompounderName"
+          : "getBalancerCompounderName",
+        args: [asset ?? "0x"],
+      },
+      {
+        ...registryContract,
+        functionName: isToken
+          ? "getTokenCompounder"
+          : isCurve
+          ? "getCurveCompounder"
+          : "getBalancerCompounder",
+        args: [asset ?? "0x"],
+      },
+      {
+        ...registryContract,
+        functionName: isToken
+          ? // dummy request that returns the same type of data
+            "getTokenCompoundersList"
+          : isCurve
+          ? "getCurveCompounderUnderlyingAssets"
+          : "getBalancerCompounderUnderlyingAssets",
+        args: [asset ?? "0x"],
+      },
+    ],
+    enabled: !!asset && enableFallback,
+  })
+
+  // Prioritize API response until it has errored
+  if (isToken && !apiTokenQuery.isError) {
+    const matchedCompounder = findApiTokenVaultForAsset(
+      apiTokenQuery.data,
+      asset
+    )
+    return {
+      ...apiTokenQuery,
+      data: {
+        name: matchedCompounder?.name,
```

Comment:
Fetch as much data from the registry in one `multicall` as possible + return normalized data used by both compounders and concentrators.